tests/AutoReview/ProjectFixerConfigurationTest.php

```diff
@@ -26,23 +28,46 @@
  */
 final class ProjectFixerConfigurationTest extends TestCase
 {
-    public function testCreate()
+    /**
+     * @var \PhpCsFixer\Config
+     */
+    private $config;
+
+    protected function setUp()
     {
-        /** @var \PhpCsFixer\Config $config */
-        $config = require __DIR__.'/../../.php_cs.dist';
+        $file = __DIR__.'/../../.php_cs.dist';
+        $this->assertFileExists($file);
+        $this->config = require $file;
+        $this->assertInstanceOf(Config::class, $this->config);
+    }
 
-        $this->assertInstanceOf(\PhpCsFixer\Config::class, $config);
-        $this->assertEmpty($config->getCustomFixers());
-        $this->assertNotEmpty($config->getRules());
+    public function testCreate()
+    {
+        $this->assertEmpty($this->config->getCustomFixers());
+        $this->assertNotEmpty($this->config->getRules());
 
         // call so the fixers get configured to reveal issue (like deprecated configuration used etc.)
         $resolver = new ConfigurationResolver(
-            $config,
+            $this->config,
             [],
             __DIR__,
             new ToolInfo()
         );
 
-        $resolver->getFixers();
+        $fixers = $resolver->getFixers();
+
+        $this->assertCount(count($this->config->getRules()), $fixers);
+    }
+
+    public function testRuleDefinedAlpha()
+    {
+        $rules = $rulesSorted = array_keys($this->config->getRules());
+        natcasesort($rulesSorted);
+
+        $this->assertSame(
+            $rulesSorted,
+            $rules,
+            (new Differ())->diff(var_export($rulesSorted, true), var_export($rules, true))
```

Comment:
isn't failing assertion provides the diff already ?

Reply:
it kind of does (ignore the first failure);
```
$ ./vendor/bin/phpunit tests/AutoReview/ProjectFixerConfigurationTest.php
PHPUnit 6.5.7 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.2.3-1+ubuntu16.04.1+deb.sury.org+1
Configuration: /home/possum/work/PHP-CS-Fixer/phpunit.xml.dist

Testing PhpCsFixer\Tests\AutoReview\ProjectFixerConfigurationTest
FF                                                                                                                                                                  2 / 2 (100%)

Time: 112 ms, Memory: 10.00MB

There were 2 failures:

1) PhpCsFixer\Tests\AutoReview\ProjectFixerConfigurationTest::testCreate
Failed asserting that actual size 151 matches expected size 50.

/home/possum/work/PHP-CS-Fixer/tests/AutoReview/ProjectFixerConfigurationTest.php:59

2) PhpCsFixer\Tests\AutoReview\ProjectFixerConfigurationTest::testRuleDefinedAlpha
Failed asserting that Array &0 (
    0 => '@PHP56Migration'
    1 => '@PHPUnit60Migration:risky'
    2 => '@Symfony'
    3 => '@Symfony:risky'
    4 => 'align_multiline_comment'
    5 => 'array_indentation'
    6 => 'array_syntax'
    7 => 'blank_line_before_statement'
    8 => 'combine_consecutive_issets'
    9 => 'combine_consecutive_unsets'
    10 => 'comment_to_phpdoc'
    11 => 'compact_nullable_typehint'
    12 => 'escape_implicit_backslashes'
    13 => 'explicit_indirect_variable'
    14 => 'explicit_string_variable'
    15 => 'final_internal_class'
    16 => 'fully_qualified_strict_types'
    17 => 'function_to_constant'
    18 => 'header_comment'
    19 => 'heredoc_to_nowdoc'
    20 => 'list_syntax'
    21 => 'method_argument_space'
    22 => 'method_chaining_indentation'
    23 => 'multiline_comment_opening_closing'
    24 => 'no_alternative_syntax'
    25 => 'no_extra_blank_lines'
    26 => 'no_null_property_initialization'
    27 => 'no_short_echo_tag'
    28 => 'no_superfluous_elseif'
    29 => 'no_unneeded_curly_braces'
    30 => 'no_unneeded_final_method'
    31 => 'no_unreachable_default_argument_value'
    32 => 'no_useless_else'
    33 => 'no_useless_return'
    34 => 'ordered_class_elements'
    35 => 'ordered_imports'
    36 => 'php_unit_ordered_covers'
    37 => 'php_unit_set_up_tear_down_visibility'
    38 => 'php_unit_strict'
    39 => 'php_unit_test_annotation'
    40 => 'php_unit_test_class_requires_covers'
    41 => 'phpdoc_add_missing_param_annotation'
    42 => 'phpdoc_order'
    43 => 'phpdoc_types_order'
    44 => 'semicolon_after_instruction'
    45 => 'single_line_comment_style'
    46 => 'strict_comparison'
    47 => 'strict_param'
    48 => 'string_line_ending'
    49 => 'yoda_style'
) is identical to Array &0 (
    0 => '@PHP56Migration'
    1 => '@PHPUnit60Migration:risky'
    2 => '@Symfony'
    3 => '@Symfony:risky'
    4 => 'align_multiline_comment'
    5 => 'array_indentation'
    6 => 'array_syntax'
    7 => 'blank_line_before_statement'
    8 => 'combine_consecutive_issets'
    9 => 'combine_consecutive_unsets'
    10 => 'comment_to_phpdoc'
    11 => 'compact_nullable_typehint'
    12 => 'escape_implicit_backslashes'
    13 => 'explicit_indirect_variable'
    14 => 'explicit_string_variable'
    15 => 'final_internal_class'
    16 => 'fully_qualified_strict_types'
    17 => 'function_to_constant'
    18 => 'header_comment'
    19 => 'heredoc_to_nowdoc'
    20 => 'list_syntax'
    21 => 'method_argument_space'
    22 => 'method_chaining_indentation'
    23 => 'multiline_comment_opening_closing'
    24 => 'no_alternative_syntax'
    25 => 'no_extra_blank_lines'
    26 => 'no_null_property_initialization'
    27 => 'no_short_echo_tag'
    28 => 'no_superfluous_elseif'
    29 => 'no_unneeded_curly_braces'
    30 => 'no_unneeded_final_method'
    31 => 'no_unreachable_default_argument_value'
    32 => 'no_useless_else'
    33 => 'no_useless_return'
    34 => 'ordered_class_elements'
    35 => 'ordered_imports'
    41 => 'phpdoc_add_missing_param_annotation'
    42 => 'phpdoc_order'
    43 => 'phpdoc_types_order'
    36 => 'php_unit_ordered_covers'
    37 => 'php_unit_set_up_tear_down_visibility'
    38 => 'php_unit_strict'
    39 => 'php_unit_test_annotation'
    40 => 'php_unit_test_class_requires_covers'
    44 => 'semicolon_after_instruction'
    45 => 'single_line_comment_style'
    46 => 'strict_comparison'
    47 => 'strict_param'
    48 => 'string_line_ending'
    49 => 'yoda_style'
).

/home/possum/work/PHP-CS-Fixer/tests/AutoReview/ProjectFixerConfigurationTest.php:69
```

that diff is kind of hard to read, we can drop the Differ when this PR is close to finished

Reply:
ok, let us keep it